File: hw/Pérez/U3/HW28AbstractFactorySingleton/Singleton/src/main/java/ec/edu/espe/singleton/model/USTax.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.singleton.model;

/**
 *
 * @author Carlos Perez,The Javas, DCCO-ESPE
 */
public class USTax {
    private static USTax instance;
    private USTax(){
        
    }
    public static USTax getInstance(){
        if(instance==null){
            instance = new USTax();
        }
        return instance;
    }
    public float salesTotal(){
        return 12.23f;
    }
}

File: hw/Pérez/U3/HW28AbstractFactorySingleton/SingletonJS/model/USTax.js
export class USTax {
    static instance = null;

    constructor() {
        if (USTax.instance) {
            return USTax.instance;
        }
        USTax.instance = this;
    }

    static getInstance() {
        if (!USTax.instance) {
            USTax.instance = new USTax();
        }
        return USTax.instance;
    }

    salesTotal() {
        return 12.23;
    }
}
